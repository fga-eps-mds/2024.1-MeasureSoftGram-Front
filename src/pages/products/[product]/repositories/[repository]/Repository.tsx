import React, { useState } from 'react';

import { NextPageWithLayout } from '@pages/_app.next';

import { Box, Container, Tooltip } from '@mui/material';
import SsidChartIcon from '@mui/icons-material/SsidChart';
import LineAxisIcon from '@mui/icons-material/LineAxis';
import SpeedIcon from '@mui/icons-material/Speed';
import TableRowsIcon from '@mui/icons-material/TableRows';

import { AiOutlineRadarChart } from 'react-icons/ai';

import useRequireAuth from '@hooks/useRequireAuth';

import GraphicChart from '@components/GraphicChart';
import LatestValueTable from '@components/LatestValueTable';

import Layout from '@components/Layout/Layout';
import ProductConfigFilterProvider from '@contexts/ProductConfigFilterProvider/ProductConfigFilterProvider';
import { useRepositoryContext } from '@contexts/RepositoryProvider';
import { useTranslation } from 'react-i18next';
import TreeViewFilter from './components/TreeViewFilter';
import Headers from './components/Header';
import CustomTabs from './components/CustomTabs';

import { useQuery } from './hooks/useQuery';
import OptionsHeader from './components/OptionsHeader/OptionsHeader';
import TsqmiBadge from './components/TsqmiBadge';


const Repository: NextPageWithLayout = () => {
  const { latestTSQMI, latestTSQMIBadgeUrl } = useRepositoryContext();

  useRequireAuth();
  useQuery();

  const [isHistoricCharacteristicOpen, setIsHistoricCharacteristicOpen] = useState(true);
  const [isHistoricSubCharacteristicOpen, setIsHistoricSubCharacteristicOpen] = useState(true);
  const [isHistoricMeasureOpen, setIsHistoricMeasureOpen] = useState(true);
  const { t } = useTranslation('repositories');

  const containerRef = React.useRef<HTMLDivElement>(null);

  return (
    <Box display="flex" width="100%" flexDirection="row" marginTop="40px" marginBottom="24px">
      <Container ref={containerRef} sx={{ marginBottom: '150px' }}>
        <Box marginX="1%" maxWidth="98%">
          <Headers />

          <TsqmiBadge
            latestTSQMI={latestTSQMI}
            latestTSQMIBadgeUrl={latestTSQMIBadgeUrl}
          />

          <OptionsHeader
            title={t('repository.characteristic')}
            isHistoricOpen={isHistoricCharacteristicOpen}
            setIsHistoricOpen={setIsHistoricCharacteristicOpen}
          />
          {
            isHistoricCharacteristicOpen ?
              <CustomTabs
                tabId="tab1"
                orientation="vertical"
                tabHeaderItems={[
                  <SsidChartIcon key="tab1-0" sx={{ fontSize: '21px' }} />,
                  <LineAxisIcon key="tab1-1" sx={{ fontSize: '21px' }} />
                ]}
                tabPanelItems={[
                  <GraphicChart key="tab1-0-0" title={t('repository.history')} type="msg" value="characteristics" />,
                  <GraphicChart
                    key="tab1-1-1"
                    title={t('repository.history')}
                    type="line"
                    value="characteristics"
                    addHistoricalTSQMI
                  />
                ]}
              />
              :
              <CustomTabs
                tabId="tab2"
                orientation="vertical"
                tabHeaderItems={[
                  <SpeedIcon key="tab2-0" sx={{ fontSize: '21px' }} />,
                  <AiOutlineRadarChart key="tab2-1" fontSize="22px" />,
                  <TableRowsIcon key="tab2-2" sx={{ fontSize: '21px' }} />,
                ]}
                tabPanelItems={[
                  <GraphicChart
                    key="tab2-0-0"
                    title={t("repository.actual-scenario")}
                    type="gauge"
                    autoGrid
                    value="characteristics"
                    valueType="latest-values"
                    addCurrentGoal
                  />,
                  <GraphicChart
                    key="tab2-0-1"
                    title={t("repository.actual-scenario")}
                    type="radar"
                    value="characteristics"
                    valueType="latest-values"
                    addCurrentGoal
                  />,
                  <LatestValueTable
                    key="tab2-0-2"
                    title={t('repository.history')}
                    value="characteristics"
                  />
                ]}
              />

          }

          <OptionsHeader
            title={t('repository.sub-characteristic')}
            isHistoricOpen={isHistoricSubCharacteristicOpen}
            setIsHistoricOpen={setIsHistoricSubCharacteristicOpen}
          />
          {
            isHistoricSubCharacteristicOpen ?
              <CustomTabs
                tabId="tab1"
                orientation="vertical"
                tabHeaderItems={[
                  <LineAxisIcon key="tab1-0" sx={{ fontSize: '21px' }} />
                ]}
                tabPanelItems={[
                  <GraphicChart key="tab1-0-0" title={t('repository.sub-characteristic')} type="line" value="subcharacteristics" />
                ]}
              />
              :
              <CustomTabs
                tabId="tab2"
                orientation="vertical"
                tabHeaderItems={[
                  <SpeedIcon key="tab2-0" sx={{ fontSize: '21px' }} />,
                  <AiOutlineRadarChart key="tab2-1" fontSize="22px" />,
                  <TableRowsIcon key="tab2-2" sx={{ fontSize: '21px' }} />
                ]}
                tabPanelItems={[
                  <GraphicChart
                    key="tab2-0-0"
                    title={t('repository.sub-characteristic-sub')}
                    type="gauge"
                    autoGrid
                    value="subcharacteristics"
                    valueType="latest-values"
                    addCurrentGoal
                  />,
                  <GraphicChart
                    key="tab2-0-1"
                    title={t('repository.sub-characteristic-sub')}
                    type="radar"
                    value="subcharacteristics"
                    valueType="latest-values"
                    addCurrentGoal
                  />,
                  <LatestValueTable
                    key="tab2-0-2"
                    title={t('repository.sub-characteristic')}
                    value="subcharacteristics"
                  />
                ]}
              />

          }

          <OptionsHeader
            title={t('repository.measure')}
            isHistoricOpen={isHistoricMeasureOpen}
            setIsHistoricOpen={setIsHistoricMeasureOpen}
          />
          {
            isHistoricMeasureOpen ?
              <CustomTabs
                tabId="tab1"
                orientation="vertical"
                tabHeaderItems={[
                  <LineAxisIcon key="tab1-0" sx={{ fontSize: '21px' }} />,
                ]}
                tabPanelItems={[
                  <GraphicChart key="tab1-0-0" title={t('repository.measure')} type="line" value="measures" />
                ]}

              />
              :
              <CustomTabs
                tabId="tab2"
                orientation="vertical"
                tabHeaderItems={[
                  <SpeedIcon key="tab2-0" sx={{ fontSize: '21px' }} />,
                  <AiOutlineRadarChart key="tab2-1" fontSize="22px" />,
                  <TableRowsIcon key="tab2-2" sx={{ fontSize: '21px' }} />

                ]}
                tabPanelItems={[
                  <GraphicChart
                    key="tab2-0-0"
                    title={t('repository.measure-scenario')}
                    type="gauge"
                    autoGrid
                    value="measures"
                    valueType="latest-values"
                    addCurrentGoal
                  />,
                  <GraphicChart
                    key="tab2-0-1"
                    title={t('repository.measure-scenario')}
                    type="radar"
                    value="measures"
                    valueType="latest-values"
                    addCurrentGoal
                  />,
                  <LatestValueTable
                    key="tab2-0-2"
                    title={t('repository.measure')}
                    value="measures"
                  />
                ]}
              />
          }

          <Box
            display="flex"
            flexDirection="row"
            height={60}
            alignItems="center"
          >
            <h2 style={{ color: '#113D4C', fontWeight: '500', fontSize: '25px' }}>{t('repository.measure-history')}</h2>
          </Box>

          <GraphicChart key="sonargraph-measure" title="Sonar" type="line" value="measures" collectionSource='sonarqube' />

          <GraphicChart
            key="ghGraph-measure"
            title="Github"
            type="line"
            value="measures"
            collectionSource='github'
          />

          <Box
            display="flex"
            flexDirection="row"
            height={60}
            alignItems="center"
          >
            <h2 style={{ color: '#113D4C', fontWeight: '500', fontSize: '25px' }}>{t('repository.measure-history')}</h2>
          </Box>

          <GraphicChart key="sonargraph" title="Sonar" type="line" value="metrics" collectionSource='sonarqube' />

          <GraphicChart
            key="ghGraph"
            title="Github"
            type="line"
            value="metrics"
            collectionSource='github'
          />

          <Box marginY="12px">
            <LatestValueTable title={t('repository.metrics')} value="metrics" />
          </Box>
        </Box >
      </Container >
    </Box >
  );
};

Repository.getLayout = function getLayout(page) {
  return (
    <ProductConfigFilterProvider>
      <Layout rightSide={<TreeViewFilter />}> {page}</Layout >
    </ProductConfigFilterProvider >
  );
}

export default Repository;
