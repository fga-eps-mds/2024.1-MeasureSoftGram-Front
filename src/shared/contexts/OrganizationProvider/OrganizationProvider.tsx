import React, { createContext, useState, useContext, ReactNode, useMemo, useEffect } from 'react';
import { Organization } from '@customTypes/organization';
import { organizationQuery } from '@services/organization';
import { toast } from 'react-toastify';
import { useAuth } from '@contexts/Auth';

interface Props {
  children: ReactNode;
}

interface IOrganizationContext {
  currentOrganization: Organization | null;
  currentOrganizations: Organization[];
  setCurrentOrganizations: (organizations: Organization[]) => void;
  organizationList: Organization[];
  isLoading: boolean;
  fetchOrganizations: (forceFetch?: boolean) => void;
}

const OrganizationContext = createContext<IOrganizationContext | undefined>(undefined);

export function OrganizationProvider({ children }: Props) {
  const { session } = useAuth();
  const [currentOrganizations, setCurrentOrganizations] = useState<Organization[]>([]);
  const [currentOrganization, setCurrentOrganization] = useState<Organization | null>(null);
  const [organizationList, setOrganizationList] = useState<Organization[]>([]);
  const [isLoading, setIsLoading] = useState(true);

  const fetchOrganizations = async (forceFetch?: boolean) => {
    if (!session && !forceFetch) return;
    setIsLoading(true);
    try {
      const result = await organizationQuery.getAllOrganization();
      if (result.type === 'success') {
        const organizations = result.value.map(org => ({
          id: org.id ?? '',
          name: org.name,
          description: org.description ?? '',
          url: org.url ?? '',
          products: org.products ?? [],
          key: org.key ?? ''
        }));
        setOrganizationList(organizations);
      } else {
        toast.error("Erro ao carregar organizações.");
      }
    } catch (error) {
      console.error("Failed to fetch organizations:", error);
      toast.error("Erro ao carregar organizações. Por favor, tente novamente.");
    } finally {
      setIsLoading(false);
    }
  };

  useEffect(() => {
    fetchOrganizations();
  }, []);

  useEffect(() => {
    if (organizationList.length > 0 && currentOrganizations.length === 0) {
      setCurrentOrganizations([organizationList[0]]);
    }
  }, [organizationList, currentOrganizations]);

  useEffect(() => {
    if (currentOrganizations.length > 0) {
      setCurrentOrganization(currentOrganizations[0]);
    } else {
      setCurrentOrganization(null);
    }
  }, [currentOrganizations]);

  const value = useMemo(() => {
    return {
      currentOrganization,
      currentOrganizations,
      setCurrentOrganizations,
      organizationList,
      isLoading,
      fetchOrganizations
    };
  }, [currentOrganization, currentOrganizations, organizationList, isLoading]);

  return <OrganizationContext.Provider value={value}>{children}</OrganizationContext.Provider>;
}

export function useOrganizationContext() {
  const context = useContext(OrganizationContext);

  if (context === undefined) {
    throw new Error('OrganizationContext must be used within a OrganizationProvider');
  }

  return context;
}
