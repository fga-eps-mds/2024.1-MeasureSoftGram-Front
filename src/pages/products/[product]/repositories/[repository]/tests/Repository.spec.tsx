import React from 'react';
import { render } from '@testing-library/react';

import { OrganizationProvider } from '@contexts/OrganizationProvider';
import Repository from '../Repository';

jest.mock('@contexts/ProductProvider', () => ({
  useProductContext: () => ({
    currentProduct: {
      name: `aoba`
    }
  })
}));

jest.mock('@contexts/RepositoryProvider', () => ({
  useRepositoryContext: () => ({})
}));

jest.mock('next/router', () => ({
  useRouter: () => ({
    push: () => jest.fn()
  })
}));

describe('<Repository />', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <OrganizationProvider>
        <Repository />
      </OrganizationProvider>
    );

    Array.from(container.getElementsByClassName('echarts-for-react')).forEach((chart) => {
      chart.setAttribute('_echarts_instance_', 'ec_123');
    });

    expect(container).toMatchSnapshot();
  });
});
