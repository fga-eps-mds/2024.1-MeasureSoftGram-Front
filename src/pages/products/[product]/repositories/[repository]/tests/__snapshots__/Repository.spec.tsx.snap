// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repository /> Should render correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-vb578k"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-mednve-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-bbsvb3"
      >
        <div
          class="MuiBox-root css-9jay18"
        >
          <div
            class="MuiBox-root css-j7qwjs"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-obdthj-MuiTypography-root"
              >
                Reposit√≥rio
              </h4>
              <h4
                class="MuiTypography-root MuiTypography-h4 css-r2wbd1-MuiTypography-root"
              />
            </div>
            <span
              class="MuiTypography-root MuiTypography-caption css-qkcnm0-MuiTypography-root"
            />
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SettingsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ContentCopyIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div />
        <div
          class="MuiBox-root css-1ntb6q6"
        >
          <h2
            style="color: rgb(17, 61, 76); font-weight: 500; font-size: 25px;"
          >
            repository.characteristic
          </h2>
          <div
            class="MuiBox-root css-1vr4rvm"
            style="width: 50%;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              history
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              current-scenario
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-1880h6w"
        >
          <div
            class="MuiTabs-root MuiTabs-vertical css-tf8uax-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
              style="overflow: hidden; margin-right: 0px;"
            >
              <div
                aria-orientation="vertical"
                class="MuiTabs-flexContainer MuiTabs-flexContainerVertical css-lfwcke-MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="vertical-tab1panel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-bavzl3-MuiButtonBase-root-MuiTab-root"
                  id="vertical-tab1-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lydjla-MuiSvgIcon-root"
                    data-testid="SsidChartIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M21 5.47 12 12 7.62 7.62 3 11V8.52L7.83 5l4.38 4.38L21 3v2.47zM21 15h-4.7l-4.17 3.34L6 12.41l-3 2.13V17l2.8-2 6.2 6 5-4h4v-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="vertical-tab1panel-1"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-bavzl3-MuiButtonBase-root-MuiTab-root"
                  id="vertical-tab1-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lydjla-MuiSvgIcon-root"
                    data-testid="LineAxisIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m22 7.43-1.41-1.41-4.03 4.53L9.5 4 2 11.51l1.5 1.5 6.14-6.15 5.59 5.18-1.73 1.95-4-4L2 17.5 3.5 19l6-6.01 4 4 3.19-3.59 3.9 3.61L22 15.6l-3.98-3.7z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator css-10d9dml-MuiTabs-indicator"
                style="top: 0px; height: 0px;"
              />
            </div>
          </div>
          <div
            aria-labelledby="vertical-tab-0"
            class="MuiBox-root css-1njwyir"
            id="vertical-tabpanel-0"
            role="tabpanel"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
              style="height: 300px;"
            />
          </div>
          <div
            aria-labelledby="vertical-tab-1"
            class="MuiBox-root css-1nw0r8e"
            id="vertical-tabpanel-1"
            role="tabpanel"
          >
            <div
              class="MuiBox-root css-13kcqfm"
              style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                _echarts_instance_="ec_123"
                class="echarts-for-react "
                size-sensor-id="2"
                style="height: 300px; webkit-user-select: none; user-select: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);"
              >
                <div
                  style="position: relative; width: 0px; height: 300px; padding: 0px; margin: 0px; border-width: 0px;"
                />
                <object
                  class="size-sensor-object"
                  data="about:blank"
                  style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
                  tabindex="-1"
                  type="text/html"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1ntb6q6"
        >
          <h2
            style="color: rgb(17, 61, 76); font-weight: 500; font-size: 25px;"
          >
            repository.sub-characteristic
          </h2>
          <div
            class="MuiBox-root css-1vr4rvm"
            style="width: 50%;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              history
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              current-scenario
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-1880h6w"
        >
          <div
            class="MuiTabs-root MuiTabs-vertical css-tf8uax-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
              style="overflow: hidden; margin-right: 0px;"
            >
              <div
                aria-orientation="vertical"
                class="MuiTabs-flexContainer MuiTabs-flexContainerVertical css-lfwcke-MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="vertical-tab1panel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-bavzl3-MuiButtonBase-root-MuiTab-root"
                  id="vertical-tab1-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lydjla-MuiSvgIcon-root"
                    data-testid="LineAxisIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m22 7.43-1.41-1.41-4.03 4.53L9.5 4 2 11.51l1.5 1.5 6.14-6.15 5.59 5.18-1.73 1.95-4-4L2 17.5 3.5 19l6-6.01 4 4 3.19-3.59 3.9 3.61L22 15.6l-3.98-3.7z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator css-10d9dml-MuiTabs-indicator"
                style="top: 0px; height: 0px;"
              />
            </div>
          </div>
          <div
            aria-labelledby="vertical-tab-0"
            class="MuiBox-root css-1njwyir"
            id="vertical-tabpanel-0"
            role="tabpanel"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
              style="height: 300px;"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-1ntb6q6"
        >
          <h2
            style="color: rgb(17, 61, 76); font-weight: 500; font-size: 25px;"
          >
            repository.measure
          </h2>
          <div
            class="MuiBox-root css-1vr4rvm"
            style="width: 50%;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              history
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              current-scenario
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-1880h6w"
        >
          <div
            class="MuiTabs-root MuiTabs-vertical css-tf8uax-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
              style="overflow: hidden; margin-right: 0px;"
            >
              <div
                aria-orientation="vertical"
                class="MuiTabs-flexContainer MuiTabs-flexContainerVertical css-lfwcke-MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="vertical-tab1panel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-bavzl3-MuiButtonBase-root-MuiTab-root"
                  id="vertical-tab1-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lydjla-MuiSvgIcon-root"
                    data-testid="LineAxisIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m22 7.43-1.41-1.41-4.03 4.53L9.5 4 2 11.51l1.5 1.5 6.14-6.15 5.59 5.18-1.73 1.95-4-4L2 17.5 3.5 19l6-6.01 4 4 3.19-3.59 3.9 3.61L22 15.6l-3.98-3.7z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator css-10d9dml-MuiTabs-indicator"
                style="top: 0px; height: 0px;"
              />
            </div>
          </div>
          <div
            aria-labelledby="vertical-tab-0"
            class="MuiBox-root css-1njwyir"
            id="vertical-tabpanel-0"
            role="tabpanel"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
              style="height: 300px;"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-jzh824"
        >
          <h2
            style="color: rgb(17, 61, 76); font-weight: 500; font-size: 25px;"
          >
            repository.measure-history
          </h2>
        </div>
        <div
          class="MuiBox-root css-13kcqfm"
          style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            _echarts_instance_="ec_123"
            class="echarts-for-react "
            size-sensor-id="5"
            style="height: 300px; webkit-user-select: none; user-select: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);"
          >
            <div
              style="position: relative; width: 0px; height: 300px; padding: 0px; margin: 0px; border-width: 0px;"
            />
            <object
              class="size-sensor-object"
              data="about:blank"
              style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-13kcqfm"
          style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            _echarts_instance_="ec_123"
            class="echarts-for-react "
            size-sensor-id="6"
            style="height: 300px; webkit-user-select: none; user-select: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);"
          >
            <div
              style="position: relative; width: 0px; height: 300px; padding: 0px; margin: 0px; border-width: 0px;"
            />
            <object
              class="size-sensor-object"
              data="about:blank"
              style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-jzh824"
        >
          <h2
            style="color: rgb(17, 61, 76); font-weight: 500; font-size: 25px;"
          >
            repository.measure-history
          </h2>
        </div>
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
          style="height: 300px;"
        />
        <div
          class="MuiBox-root css-13kcqfm"
          style="opacity: 1; webkit-transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            _echarts_instance_="ec_123"
            class="echarts-for-react "
            size-sensor-id="8"
            style="height: 300px; webkit-user-select: none; user-select: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);"
          >
            <div
              style="position: relative; width: 0px; height: 300px; padding: 0px; margin: 0px; border-width: 0px;"
            />
            <object
              class="size-sensor-object"
              data="about:blank"
              style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-10khgmf"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-j70xtz-MuiSkeleton-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
